content/src/main/content/jcr_root/apps/acs-commons/cloudservices/pwa/clientlibs/clientlib-pwa/pwa.js

```diff
@@ -0,0 +1,69 @@
+(function() {
+
+    var promptInstall,
+        currentPath = window.location.pathname,
+        pathName = currentPath,
+        serviceWorkerJsPath = Granite.$("meta['name=pwa.root']").attr("value");
+
+    function addManifestToDOM(){
+        var  el, head = document.getElementsByTagName("head")[0];
+        el = document.createElement("link");
+        el.rel = "manifest";
+        el.href = pathName.replace('.html','') +'.pwa.load/manifest.json';
+        head.appendChild(el);
+    }
+
+    if ('serviceWorker' in navigator) {
+        var rootSWPath = pathName.replace('.html','')+'.pwa.load/root-service-worker.json';
+
+        Granite.$.ajax({
+            type : "GET",
+            url : rootSWPath,
+        }).then(function(data) {
+            if(data && data.path){
+                loadServiceWorker(data.path);
+            }  else {
+                console.error('Failed to load from PWA Service Worker.');
+            }
+        }, function(error){
+            console.error('Failed to load PWA Service Worker', error);
+        });
+
+    }
+
+    function loadServiceWorker(data){
+        navigator.serviceWorker
+            .register(serviceWorkerJsPath)
```

Comment:
@davidjgonzalez  This looks good. Lemme try on the env. But the JS files are outdated